src/main/java/com/feedzai/commons/sql/abstraction/engine/impl/MySqlTranslator.java

```diff
@@ -286,6 +287,14 @@ public String translate(final With with) {
         throw new OperationNotSupportedRuntimeException("MySQL does not support WITH.");
     }
 
+    @Override
+    protected Union rowsToUnion(final List<Expression> rows) {
+        // By default, use UNION ALL to express VALUES.
+        // MySQL does not support modeling UNION using binary tree due to a bug prior to 8.0
+        // https://bugs.mysql.com/bug.php?id=25734
+        return union(rows).all();
```

Comment:
Does MySQL break with lots of values?

Reply:
MySQL has a bug regarding parenthesis parsing when in the presence of union.
It was fixed in the last release but PDB does not support it.